### PR TITLE
Use str to get the exception message

### PR DESCRIPTION
**Pull request by stsewd**
We are catching exceptions of type `Exception`, which no always have the `message` attribute (python3 at least).

```python
e = Exception('Bu!')
assert not hasattr(e, 'message')
```

### CODE REVIEW

**Comment by stsewd** [comment]
These class don't override the `init` method, so everything is ok

https://github.com/rtfd/readthedocs.org/blob/2c9e62d7458e7d17ea121226fb45c9e57c55c6cd/readthedocs/vcs_support/utils.py#L15-L16

https://github.com/rtfd/readthedocs.org/blob/2c9e62d7458e7d17ea121226fb45c9e57c55c6cd/readthedocs/core/views/hooks.py#L22-L23

https://github.com/rtfd/readthedocs.org/blob/2c9e62d7458e7d17ea121226fb45c9e57c55c6cd/readthedocs/projects/exceptions.py#L52-L61

https://github.com/rtfd/readthedocs.org/blob/2c9e62d7458e7d17ea121226fb45c9e57c55c6cd/readthedocs/core/views/__init__.py#L31-L32

These override the `init` method correctly

https://github.com/rtfd/readthedocs.org/blob/2c9e62d7458e7d17ea121226fb45c9e57c55c6cd/readthedocs/core/utils/tasks/retrieve.py#L10-L13

https://github.com/rtfd/readthedocs.org/blob/2c9e62d7458e7d17ea121226fb45c9e57c55c6cd/readthedocs/core/utils/tasks/public.py#L103-L107

I found those grepping the project.

**Comment by stsewd** [comment]
@berkerpeksag I take the `import` from http://python-future.org/compatible_idioms.html#unicode

I check for an exception with non-ascii characters, this is what a found

```
>>> e = Exception('pitón')
>>> str(e)
'pit\xc3\xb3n'
>>> e.message
'pit\xc3\xb3n'
>>> from builtins import str
>>> e = Exception('pitón')
>>> str(e)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/site-packages/future/types/newstr.py", line 102, in __new__
    return super(newstr, cls).__new__(cls, value)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 3: ordinal not in range(128)
```

With the imported str an exception happens. So, are we fine just removing the import?

**Comment by berkerpeksag** [comment]
Ah, so `builtins` is coming from the future module. I forgot that there is no `builtins` module in Python 2. It has been a while since I wrote code in Python 2 :)

I think using future and their implicit imports to do this is a bit bad idea. Even without using its implicit conversation, it's possible to get a `UnicodeEncodeError`:

```py
>>> e = Exception(u'ıçğü')
>>> str(e)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-3: ordinal not in range(128)
```

(Since Django and other libraries use unicode to store data internally, the chance of getting a unicode `exc.message` (`u'ı'` for example) is more than a str (`'ı'` for example)

I'd be more conservative and don't do this until we get rid of Python 2, but of course this is just my opinion. If we decide to get this in, it would be nice to check if we reach 100% branch coverage (including adding tests for the `u'non-ascıı'` case) for both usages of `str(exc)`.

**Comment by stsewd** [comment]
Maybe put this for python 3 milestone is the best

File: readthedocs/restapi/views/model_views.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
     absolute_import, division, print_function, unicode_literals)
 
 import logging
+from builtins import str
```
Comment (agjohnson) [inline readthedocs/restapi/views/model_views.py]: `builtins` is third party import

**Comment by agjohnson** [comment]
We're targeting 3.6 swap in prod in the move to new host. I'll target this at 2.5 milestone for now.

**Comment by stsewd** [comment]
@agjohnson 

> it seems we're disregarding some of our use cases of exceptions here.
> More specifically, exceptions that use message.

Are you referring to a code that uses `e.message`? All custom exceptions that I found pass the `message` attribute to their parent `init` method (till Exception), so `e.message` == `str(e)`.

At least an exception change the `e.message` attribute after instantiating the object.

**Comment by agjohnson** [comment]
Prod is 3.6 now, probably time to merge this!

**Pull request merged**